src/ophys_etl/transforms/event_detection.py

```diff
@@ -0,0 +1,571 @@
+                        "DFF traces < April, 2019. Consider recreating the "
+                        "DFF file with a current version of that module.")
+        return data
+
+    @mm.post_load
+    def get_noise_multiplier(self, data, **kwargs):
+        if 'noise_multiplier' in data:
+            return data
+        val_11hz = 3.0
+        val_31hz = 1.0
+        if np.round(data['movie_frame_rate_hz'] / 11.0) == 1:
+            data['noise_multiplier'] = val_11hz
+        elif np.round(data['movie_frame_rate_hz'] / 31.0) == 1:
+            data['noise_multiplier'] = val_31hz
+        else:
+            raise EventDetectionException(
+                "You did not specify 'noise_multiplier'. In that case, the "
+                "multiplier is selected by 'movie_frame_rate_hz' but only "
+                "freqeuncies around 11Hz and 31Hz have specified values: "
+                f"({val_11hz}, {val_31hz}) respectively. You specified a "
+                f"'movie_frame_rate' of {data['movie_frame_rate_hz']}")
+        return data
+
+    @mm.post_load
+    def cpu_to_the_max(self, data, **kwargs):
+        if data['n_parallel_workers'] == -1:
+            data['n_parallel_workers'] = multiprocessing.cpu_count()
+        return data
+
+    @mm.post_load
+    def get_decay_time(self, data, **kwargs):
+        if ('full_genotype' not in data) & ('decay_time' not in data):
+            raise EventDetectionException(
+                    "Must provide either `decay_time` or `full_genotype` "
+                    "for decay time lookup. "
+                    "Available lookup values are "
+                    f"\n{json.dumps(decay_lookup, indent=2)}")
+        if 'full_genotype' in data:
+            if data['full_genotype'] not in decay_lookup:
+                raise EventDetectionException(
+                        f"{data['full_genotype']} not available. "
+                        "Available lookup values are "
+                        f"\n{json.dumps(decay_lookup, indent=2)}")
+            lookup = decay_lookup[data['full_genotype']]
+            if 'decay_time' in data:
+                warnings.warn("You specified a decay_time of "
+                              f"{data['decay_time']} but that is being "
+                              "overridden by a lookup by genotype to give "
+                              f"a decay time of {lookup}.")
+            data['decay_time'] = lookup
+        data['halflife'] = calculate_halflife(data['decay_time'])
+        return data
+
+
+def fast_lzero(penalty: float, dat: np.ndarray, gamma: float,
+               constraint: bool) -> np.ndarray:
+    """runs fast spike inference and returns an array like the input trace
+    with data substituted by event magnitudes
+
+    Parameters
+    ----------
+    penalty: float
+        tuning parameter lambda > 0
+    dat: np.ndarray
+        fluorescence data
+    gamma: float
+        a scalar value for the AR(1) decay parameter; 0 < gam <= 1
```

Comment:
Not exactly sure what `gam` is here. Is it different from `gamma`?

Reply:
fixed